### ISSUE TITLE
Use micro & micro-dev programmatically?

### ISSUE THREAD

**Issue by track0x1** [vercel/micro#337]
Per the main README:
> You can use Micro programmatically by requiring Micro directly

Is there a way to use both micro and micro-dev programmatically?

**Comment by Callum-Newton** [comment]
There's not currently a way to use micro-dev programmatically afaict, how would you want to use it?

**Comment by track0x1** [comment]
```js
const micro = require('micro');
```

and perhaps if running with a development flag it will provide `micro-dev` instead?

**Comment by fmiras** [comment]
Since this feature isn't available until micro-dev supports programmatically mode, I made this workaround to achieve what you want to do:

```json
// package.json
{
  "name": "micro-dev-proof",
  "description": "Test of micro-dev programatically usage",
  "main": "index.js",
  "scripts": {
    "start": "NODE_ENV=production node .",
    "dev": "node ."
  },
  "license": "MIT",
  "dependencies": {
    "micro": "9.1.0"
  },
  "devDependencies": {
    "micro-dev": "2.2.0"
  }
}
```

```javascript
// index.js
const micro = require('./micro')
micro('./server.js')
```

```javascript
// micro.js
const path = require('path')

const micro = (handlerRelativePath) => {
  const handlerPath = path.join(process.cwd(), handlerRelativePath)

  if (process.env.NODE_ENV === 'production') {
    const handler = require(handlerPath)
    const micro = require('micro')
    const server = micro(handler)
    server.listen(3000)
  } else {
    // Don't load packages if prod
    const flags = { silent: false, host: '::', port: 3000, limit: '1mb', _: {} }
    const micro = require('micro-dev/lib/serve')
    micro(handlerPath, flags)
  }
}

module.exports = micro
```

```javascript
// server.js
module.exports = (req, res) => {
  return 'Hello world'
}
```

Full example on this repo: https://github.com/fmiras/micro-dev-proof

I hope this help works for you @track0x1

**Comment by timneutkens** [comment]
Fixed in #89
